packages/extension-ui/src/Popup/Contacts/AddContact.tsx

```diff
@@ -0,0 +1,269 @@
+// Copyright 2019-2021 @polkadot/extension-ui authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import type { ThemeProps } from '../../types';
+
+import queryString from 'query-string';
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+import { RouteComponentProps } from 'react-router';
+import styled from 'styled-components';
+
+import { Contact } from '@polkadot/extension-base/background/types';
+import { ContactsStore } from '@polkadot/extension-base/stores';
+import { decodeAddress, encodeAddress } from '@polkadot/keyring';
+import { hexToU8a, isHex } from '@polkadot/util';
+
+import { ActionBar, ActionContext, ActionText, Button, InputWithLabel } from '../../components';
+import useTranslation from '../../hooks/useTranslation';
+import { Header } from '../../partials';
+import chains from '../../util/chains';
+
+const bs58 = require('bs58');
+
+/**
+ * Get address prefix
+ * Polkadot: 0
+ * Kusama: 2
+ * Substrate: 42
+ * others: ...
+ * @param address
+ */
+function getAddressPrefix (address: string): number {
+  const bytes = bs58.decode(address);
+  const hex = bytes.toString('hex');
+  const prefix = `${hex[0]}${hex[1]}`;
+
+  return parseInt(prefix, 16);
+}
+
+/**
+ * Check address is valid address
+ * @param address
+ */
+function isValidAddressPolkadotAddress (address: string): boolean {
```

Comment:
https://github.com/polkadot-js/common/blob/master/packages/util-crypto/src/address/is.ts

(That one is actually new, generally the check was exactly this previously - although the recommendation has always been to just decode)